modules/@apostrophecms/doc-type/index.js

```diff
@@ -931,6 +854,108 @@ module.exports = {
         } else {
           return self.apos.i18n.inferIdLocaleAndMode(req, _id);
         }
+      },
+
+      // Returns an object containing the properties of doc that
+      // require deduplication when archived, stored as
+      // "previous published" or any other scenario where the slug
+      // and similar properties should never be treated as "in conflict"
+      // with content that is in play on the site. The returned object
+      // contains values for those properties that have been deduplicated, and
+      // can be passed to $set or Object.assign or both, depending on your
+      // situation. `doc` is not changed.
+
+      async deduplicate(req, doc) {
```

Comment:
Also no side effects in a sense, although it does have to do read operations on the database to do its job.

Reply:
I don't see any `await` in here.